### ISSUE TITLE
Fix false positives in *-no-unknown

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#5157]
Some pseudo classes/elements are missing from the whitelists of:
- selector-pseudo-element-no-unknown
- selector-pseudo-class-no-unknown

> sources

https://www.chromestatus.com/feature/5676301379698688
https://www.chromestatus.com/feature/6537562418053120
https://github.com/css-modules/css-modules

> What code is needed to reproduce the bug?

```css
::file-selector-button { }
:state() { }
```
The list is not exhaustive. Personally I think `:global` and `:local` could be considered as well.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-no-unknown": true,
  }
}
```

> Which version of stylelint are you using?

https://stylelint.io/demo

> What did you expect to happen?

no errors

**Comment by jeddy3** [comment]
@Mouvedia Thanks for the report and for using the template.

> The list is not exhaustive

We'll need to update our [reference lists of keywords](https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) this if you have time.

> Personally I think `:global` and `:local` could be considered as well.

The rules consider pseudo-classes and elements to be known. Those that belong to methodologies and language extensions can be allowed using the [`ignorePseudoClasses` and `ignorePseudoElements` options](https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown#optional-secondary-options). This lets configuration author explicit signal when they are using something outside of the specs.

**Comment by Mouvedia** [comment]
> Please consider contributing this if you have time.

@jeddy3 If I did, I would update the whitelists of selector-type-no-unknown and unit-no-unknown; is that a problem?

**Comment by jeddy3** [comment]
@Mouvedia That would be fantastic! Since first creating the sets, we've updated them piecemeal as users have reported missing items. It'd be great to give them the once over as new CSS specifications are constantly being added and it's likely the lists have fallen out of date.

The sets are either broken down by the sections of a specification (and usually accompanied by the URL to the specification in a comment beforehand) or by the needs of a particular rule.

Thank you!

**Comment by Mouvedia** [comment]
> It'd be great to give them the once over as new CSS specifications are constantly being added and it's likely the lists have fallen out of date.

Don't dream. I cannot make the arbitrate all the additions.
e.g. `:picture-in-picture` [may be renamed](https://github.com/w3c/picture-in-picture/pull/181) before it gets implemented
Hence I will just add the ones I had to add exceptions to in my cfg.

> Thank you!

Wait for the PR to thank me :)

**Comment by jeddy3** [comment]
Closed in https://github.com/stylelint/stylelint/pull/5158
